html/pages/device/overview/processors.inc.php

```diff
@@ -16,44 +16,88 @@
     echo '</div>
         <table class="table table-hover table-condensed table-striped">';
 
+    $totalPercent=0;
+
     foreach ($processors as $proc) {
         $text_descr = rewrite_entity_descr($proc['processor_descr']);
 
         // disable short hrDeviceDescr. need to make this prettier.
         // $text_descr = short_hrDeviceDescr($proc['processor_descr']);
         $percent      = $proc['processor_usage'];
-        $background   = get_percentage_colours($percent);
-        $graph_colour = str_replace('#', '', $row_colour);
+        if ($config['cpu_details_overview'] === true)
+        {
+
+            $background   = get_percentage_colours($percent);
+            $graph_colour = str_replace('#', '', $row_colour);
+
+            $graph_array           = array();
+            $graph_array['height'] = '100';
+            $graph_array['width']  = '210';
+            $graph_array['to']     = $config['time']['now'];
+            $graph_array['id']     = $proc['processor_id'];
+            $graph_array['type']   = $graph_type;
+            $graph_array['from']   = $config['time']['day'];
+            $graph_array['legend'] = 'no';
```

Comment:
I agree, it would be cleaner to have less duplicate, I'll see to consolidate everything above and just override what is needed.